src/plinder/data/splits.py

```diff
@@ -644,88 +652,240 @@ def assign_split_membership(
         Dataframe of splits
     """
     LOG.info(f"have {len(test_system_ids)} test system IDs")
-    entries["split"] = "train"
-    entries.loc[entries["system_id"].isin(test_system_ids), "split"] = "test"
-    remove_from_train = set()
-
-    for system_id in entries[entries["split"] == "test"]["system_id"]:
-        remove_from_train |= system_id_to_leakage.get(system_id, set())
+    pdb_id_to_systems = (
+        entries.groupby("entry_pdb_id")["system_id"].apply(set).to_dict()
+    )
 
+    remove_from_train = (
+        set.union(
+            *(
+                system_id_to_leakage.get(system_id, set())
+                for system_id in test_system_ids
+            )
+        )
+        - test_system_ids
+    )
     LOG.info(f"found {len(remove_from_train)} entries to remove from train")
+
+    # Add back MMS of test systems present in remove_from_train
+    quality_test_removed = (test_system_ids.union(remove_from_train)).intersection(
+        entries[entries["system_pass_validation_criteria"]]["system_id"]
+    )
+    mms_df = pd.read_parquet(
+        data_dir / "mmp/plinder_mmp_series.parquet",
+        filters=[("system_id", "in", quality_test_removed)],
+    )
+    mms_df["system_score"] = mms_df.groupby("system_id")["congeneric_id"].transform(
+        lambda x: x.nunique()
+    )
+    system_plindex = entries.drop_duplicates(subset=["system_id"]).reset_index(
+        drop=True
+    )
+    system_to_entry_pdb_id = dict(
+        zip(system_plindex["system_id"], system_plindex["entry_pdb_id"])
+    )
+    mms_df["entry_pdb_id"] = mms_df["system_id"].map(system_to_entry_pdb_id)
+    system_to_ligand_ccd_codes = dict(
+        zip(
+            system_plindex["system_id"],
+            system_plindex["system_proper_ligand_unique_ccd_codes"],
+        )
+    )
+    mms_df["system_proper_ligand_unique_ccd_codes"] = mms_df["system_id"].map(
+        system_to_ligand_ccd_codes
+    )
+    mms_df["unique_id"] = (
+        mms_df["entry_pdb_id"] + "__" + mms_df["system_proper_ligand_unique_ccd_codes"]
```

Comment:
should we better count for MMS uniqueness only when `system_proper_ligand_unique_ccd_codes` is different and ignore `entry_pdb_id` here?

Reply:
I think that makes the most sense conceptually, but would this require re-splitting?

Reply:
Captured this comment into a TODO for future enhancements